platform/cosmos/stake.go

```diff
@@ -38,15 +42,33 @@ func (p *Platform) GetValidators() (blockatlas.ValidatorPage, error) {
 }
 
 func (p *Platform) GetDetails() blockatlas.StakingDetails {
-	//TODO: Find a way to have a dynamic
 	return blockatlas.StakingDetails{
-		Reward:        blockatlas.StakingReward{Annual: 11},
+		Reward: blockatlas.StakingReward{
+			Annual: p.GetMaxAPR(),
+		},
 		MinimumAmount: blockatlas.Amount("0"),
 		LockTime:      1814400,
 		Type:          blockatlas.DelegationTypeDelegate,
 	}
 }
 
+func (p *Platform) GetMaxAPR() float64 {
+	validators, err := p.GetValidators()
+	if err != nil {
+		return defaultAnnualReward
```

Comment:
Can we remove default and return 0 instead maybe? 

Ideally not to have any hard coded values

Reply:
Login behind: if you provide wrong value to the user - it’s deceptive to a user. 0 would be safer in this case - but if it’s cached as 0, than it’s a problem

Reply:
Maybe throw an error would be better in this case

Reply:
No. We will need to change GetDetails() interface. If so, we will add Redis and check that err == nil, if we will have to err => use Redis (when we have cached valid value).

However, it is not good to change the basic interface, I cannot say that there is any other option how to handle it without making it more worse than rn.

So, I recommend returning the default apr if err, caching period is not critical enough to be an issue. Also, I cannot say that GetValidators() will return error often enough to care about it

Reply:
I agree to remove the hardcoded value. It's better to show 0 or nothing than show a magic number